packages/js/src/plugins/auctionHouseModule/updateAuctionHouse.ts

```diff
@@ -149,4 +162,55 @@ export const updateAuctionHouseBuilder = (
         key: params.instructionKey ?? 'updateAuctionHouse',
       })
     );
+
+  if (params.auctioneerAuthority) {
+    const auctioneerAuthority = params.auctioneerAuthority as PublicKey;
+    const ahAuctioneerPda = findAuctioneerPda(
+      auctionHouse.address,
+      auctioneerAuthority
+    );
+
+    const scopes = params.auctioneerScopes ?? [
+      AuthorityScope.Deposit,
+      AuthorityScope.Buy,
+      AuthorityScope.PublicBuy,
+      AuthorityScope.ExecuteSale,
+      AuthorityScope.Sell,
+      AuthorityScope.Cancel,
+      AuthorityScope.Withdraw,
+    ];
+    const instructionAccounts = {
+      auctionHouse: auctionHouse.address,
+      authority: authority.publicKey,
+      auctioneerAuthority,
+      ahAuctioneerPda,
+    };
+
+    const ahAuctioneerAccount = await metaplex
+      .rpc()
+      .getAccount(ahAuctioneerPda);
+
+    // Run Auctioneer update if it has been delegated.
+    if (auctionHouse.hasAuctioneer && ahAuctioneerAccount.exists) {
+      return builder.add({
+        instruction: createUpdateAuctioneerInstruction(instructionAccounts, {
+          // Scope resets when updating auctioneer.
+          // Need to provide the needed list every time or it will get to the default value.
+          scopes,
+        }),
```

Comment:
This is not a good developer experience because if a user tries to update something that's not auctioneer-specific, they will end up overriding the scopes.

We need to make additional checks before calling this instruction:
- We only call this instruction if **a new (different) auctioneer authority was provided** or **a new set of scopes was provided**
- If one is given without the other, we keep the existing settings for the auctioneer instance. For example, if we change the auctioneer authority without providing scopes, we should keep the same scopes as before!

For that last point, we need to keep additional data on the `AuctionHouse` model.

I've noticed on the program, we have all of that information on the Auctioneer account. See https://github.com/metaplex-foundation/metaplex-program-library/blob/cac9cfd042cc5a16f50d635074555392d3b9044e/auction-house/program/src/state.rs#L25-L30.

That means, we need to update the `AuctionHouse` model so that, when `hasAuctioneer` is `true`, we also capture the current `auctioneerAuthority` and the `auctioneerScopes`. That means, when fetching an auction house by address we need to make a `getMultipleAccounts` call with the address and the PDA at the same time so we still only make one HTTP request.

We also need to have some tests for these updating scenarios.

Reply:
I had the same approach previously, but the main problem is that I don't see the way to find auctioneerAuthority afterward. 
Let's imagine someone else uses already existing AH with Auctioneer, that one needs to know auctioneerAuthority, because there is no way to get it from the AH account.

I hope I'm wrong, but I haven't found a way to solve this problem.

Reply:
Urgh yeah, it is possible but it requires a `getProgramAccount` to get `Auctioneer` accounts filtered by `auction_house` attribute. 😢 

Okay forget about this, BUT:

When we call `update` we do know `auctioneerAuthority` which means we can get the `Auctioneer` account in order to get the current values to ensure they are not overridden when not explicitly provided.

Reply:
The main problem with the getProgramAccount is that it's a possible case to receive multiple auctioneerAuthorities for the single AH. I'm not sure it's expected behaviour, but program allows to do so.